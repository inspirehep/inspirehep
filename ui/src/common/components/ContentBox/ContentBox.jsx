import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Row, Col, Card } from 'antd';

import './ContentBox.scss';

class ContentBox extends Component {
  render() {
    const {
      title,
      leftActions,
      rightActions,
      loading,
      children,
      subTitle,
      className,
    } = this.props;

    return (
      children && (
        <div className={`__ContentBox__ h-100 ${className}`}>
          <Card className="h-100" title={title} loading={loading}>
            <div className="pa2">
              {subTitle && <h3 className="pb1">{subTitle}</h3>}
              <div>{children}</div>
            </div>
            <Row className="actions ph2" type="flex" justify="space-between">
              <Col>{leftActions}</Col>
              <Col>{rightActions}</Col>
            </Row>
          </Card>
        </div>
      )
    );
  }
}

ContentBox.propTypes = {
  leftActions: PropTypes.node,
  rightActions: PropTypes.node,
  children: PropTypes.node,
  title: PropTypes.string,
  subTitle: PropTypes.string,
  loading: PropTypes.bool,
  className: PropTypes.string,
};

ContentBox.defaultProps = {
  leftActions: null,
  rightActions: null,
  children: null,
  title: null,
  subTitle: null,
  loading: false,
  className: '',
};

export default ContentBox;
