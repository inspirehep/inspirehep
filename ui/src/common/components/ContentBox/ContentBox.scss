.__ContentBox__ {
  .ant-card-body {
    padding: 0 !important;
    min-height: 100%;
    display: flex;
    flex-direction: column;

    .actions {
      margin-top: auto;

      .anticon {
        font-size: 16px;
      }
    }
  }

  .ant-card-head {
    padding: 0 1rem;
  }

  .ant-card-loading-content {
    padding: 1rem;
  }
}
