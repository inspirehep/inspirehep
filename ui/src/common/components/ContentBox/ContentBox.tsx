import React from 'react';
import { Row, Col, Card } from 'antd';
import classNames from 'classnames';

import './ContentBox.less';

const ContentBox = ({
  title,
  leftActions,
  rightActions,
  loading,
  children,
  subTitle,
  className,
  fullHeight = true,
  smallPadding = true,
}: {
  title: string;
  leftActions: JSX.Element | JSX.Element[];
  rightActions: JSX.Element | JSX.Element[];
  loading: boolean;
  children: JSX.Element | JSX.Element[] | any;
  subTitle: string;
  className: string;
  fullHeight?: boolean;
  smallPadding?: boolean;
}) => {
  return (
    children && (
      <div
        className={classNames(
          '__ContentBox__',
          { 'h-100': fullHeight },
          className
        )}
      >
        <Card className="h-100" title={title} loading={loading}>
          <div
            className={classNames(
              { pa2: smallPadding },
              { pa3: !smallPadding },
              className
            )}
          >
            {subTitle && <h3 className="pb1">{subTitle}</h3>}
            <div>{children}</div>
          </div>
          <Row className="actions ph2" justify="space-between">
            <Col>{leftActions}</Col>
            <Col>{rightActions}</Col>
          </Row>
        </Card>
      </div>
    )
  );
};

ContentBox.defaultProps = {
  leftActions: null,
  rightActions: null,
  children: null,
  title: null,
  subTitle: null,
  loading: false,
  className: '',
};

export default ContentBox;
