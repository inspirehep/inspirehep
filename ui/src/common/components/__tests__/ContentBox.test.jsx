import React from 'react';
import { shallow } from 'enzyme';

import ContentBox from '../ContentBox';

describe('ContentBox', () => {
  it('renders ContentBox with actions and title without loading', () => {
    const wrapper = shallow(
      <ContentBox title="Jessica Jones" actions={[<h2 key="pi">PI</h2>]}>
        <div>Defenders</div>
      </ContentBox>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders ContentBox with actions and title with loading', () => {
    const wrapper = shallow(
      <ContentBox
        title="Jessica Jones"
        loading
        actions={[<h1 key="pi">PI</h1>]}
      >
        <div>Defenders</div>
      </ContentBox>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders ContentBox with actions', () => {
    const wrapper = shallow(
      <ContentBox actions={[<h2 key="pi">PI</h2>]}>
        <div>Defenders</div>
      </ContentBox>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders ContentBox without actions', () => {
    const wrapper = shallow(
      <ContentBox>
        <div>Defenders</div>
      </ContentBox>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders ContentBox with subTitle and className', () => {
    const wrapper = shallow(
      <ContentBox subTitle="Lame" className="pa3">
        <div>Defenders</div>
      </ContentBox>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('does not render ContentBox without children', () => {
    const wrapper = shallow(<ContentBox actions={[<h2 key="pi">PI</h2>]} />);
    expect(wrapper).toMatchSnapshot();
  });
});
