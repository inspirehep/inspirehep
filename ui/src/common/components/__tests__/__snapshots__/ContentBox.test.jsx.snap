// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentBox does not render ContentBox without children 1`] = `""`;

exports[`ContentBox renders ContentBox with actions 1`] = `
<div
  className="__ContentBox__ h-100 "
>
  <Card
    className="h-100"
    loading={false}
    title={null}
  >
    <div
      className="pa2"
    >
      <div>
        <div>
          Defenders
        </div>
      </div>
    </div>
    <Row
      className="actions ph2"
      gutter={0}
      justify="space-between"
      type="flex"
    >
      <Col />
      <Col />
    </Row>
  </Card>
</div>
`;

exports[`ContentBox renders ContentBox with actions and title with loading 1`] = `
<div
  className="__ContentBox__ h-100 "
>
  <Card
    className="h-100"
    loading={true}
    title="Jessica Jones"
  >
    <div
      className="pa2"
    >
      <div>
        <div>
          Defenders
        </div>
      </div>
    </div>
    <Row
      className="actions ph2"
      gutter={0}
      justify="space-between"
      type="flex"
    >
      <Col />
      <Col />
    </Row>
  </Card>
</div>
`;

exports[`ContentBox renders ContentBox with actions and title without loading 1`] = `
<div
  className="__ContentBox__ h-100 "
>
  <Card
    className="h-100"
    loading={false}
    title="Jessica Jones"
  >
    <div
      className="pa2"
    >
      <div>
        <div>
          Defenders
        </div>
      </div>
    </div>
    <Row
      className="actions ph2"
      gutter={0}
      justify="space-between"
      type="flex"
    >
      <Col />
      <Col />
    </Row>
  </Card>
</div>
`;

exports[`ContentBox renders ContentBox with subTitle and className 1`] = `
<div
  className="__ContentBox__ h-100 pa3"
>
  <Card
    className="h-100"
    loading={false}
    title={null}
  >
    <div
      className="pa2"
    >
      <h3
        className="pb1"
      >
        Lame
      </h3>
      <div>
        <div>
          Defenders
        </div>
      </div>
    </div>
    <Row
      className="actions ph2"
      gutter={0}
      justify="space-between"
      type="flex"
    >
      <Col />
      <Col />
    </Row>
  </Card>
</div>
`;

exports[`ContentBox renders ContentBox without actions 1`] = `
<div
  className="__ContentBox__ h-100 "
>
  <Card
    className="h-100"
    loading={false}
    title={null}
  >
    <div
      className="pa2"
    >
      <div>
        <div>
          Defenders
        </div>
      </div>
    </div>
    <Row
      className="actions ph2"
      gutter={0}
      justify="space-between"
      type="flex"
    >
      <Col />
      <Col />
    </Row>
  </Card>
</div>
`;
